### ISSUE TITLE
Waste too much time on blob cache operation

### ISSUE THREAD

**Issue by Connor1996** [tikv/titan#203]
When doing a long scan with 1000 kvs on master, find it wastes too much CPU time on blob cache mutex and affect scan performance a lot.

![image](https://user-images.githubusercontent.com/13497871/105829145-f1eb6580-5ffe-11eb-8365-62f48629e2df.png)

But for 4.0, the scan performance is not affected a lot(sorry for the flame graph is missing)

**Comment by yiwu-arbug** [comment]
By default each cache shard is at least 512KB, which means if cache size is set to 0, there's only 1 shard. It will of course cause mutex contention. Can you try to set cache to a more practical size and see?

**Comment by Connor1996** [comment]
Yes, but seems the mutex contention causes too much overhead. And in 4.0 branch, it doesn't have that much overhead when size == 0. I'll take a deep look in the future.
